packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -41,19 +45,51 @@ const TextStyle _kCupertinoDialogActionStyle = TextStyle(
 );
 
 const double _kCupertinoDialogWidth = 270.0;
-const BoxDecoration _kCupertinoDialogFrontFillDecoration = BoxDecoration(
-  color: Color(0xccffffff),
-);
-const BoxDecoration _kCupertinoDialogBackFill = BoxDecoration(
-  color: Color(0x77ffffff),
+const double _kAccessibilityCupertinoDialogWidth = 310.0;
+
+// _kCupertinoDialogBlurOverlayDecoration is applied to the blurred backdrop to
+// lighten the blurred image. Brightening is done to counteract the dark modal
+// barrier that appears behind the dialog. The overlay blend mode does the
+// brightening. The white color doesn't paint any white, it's just the basis
+// for the overlay blend mode.
+const BoxDecoration _kCupertinoDialogBlurOverlayDecoration = BoxDecoration(
+  color: CupertinoColors.white,
+  backgroundBlendMode: BlendMode.overlay,
 );
 
+const double _kBlurAmount = 20.0;
 const double _kEdgePadding = 20.0;
-const double _kButtonHeight = 45.0;
+const double _kMinButtonHeight = 45.0;
+const double _kMinButtonFontSize = 10.0;
+const double _kDialogCornerRadius = 12.0;
+const double _kDividerThickness = 1.0;
 
-// TODO(gspencer): This color isn't correct. Instead, we should carve a hole in
-// the dialog and show more of the background.
-const Color _kButtonDividerColor = Color(0xffd5d5d5);
+// Translucent white that is painted on top of the blurred backdrop as the
+// dialog's background color.
+const Color _kDialogColor = Color(0xC0FFFFFF);
+
+// Translucent white that is painted on top of the blurred backdrop as the
+// background color of a pressed button.
+const Color _kDialogPressedColor = Color(0x90FFFFFF);
+
+// Translucent white that is painted on top of the blurred backdrop in the
+// gap areas between the content section and actions section, as well as between
+// buttons.
+const Color _kButtonDividerColor = Color(0x40FFFFFF);
+
+// The alert dialog layout policy changes depending on whether the user is using
+// a "regular" font size vs a "large" font size. This is a spectrum. There are
+// many "regular" font sizes and many "large" font sizes. But depending on which
+// spectrum is currently being used, a dialog is laid out differently.
+//
+// Empirically, the jump from one spectrum to the other occurs at the following text
```

Comment:
the last two occurrences of "spectrum" here (lines 83 and 85) should presumably be "policy"?